.github/workflows/e2e-functional-template.yml

```diff
@@ -0,0 +1,184 @@
+        required: true
+        default: "ubuntu-22.04"
+      nightly:
+        type: boolean
+        description: "True if this is nigtly build"
+        required: false
+        default: false
+    outputs:
+      COMMENT_BODY:
+        description: "The output to comment"
+        value: ${{ jobs.e2e.outputs.COMMENT_BODY }}
+
+env:
+  AWS_S3_BUCKET: "mattermost-cypress-report"
+  BRANCH: ${{ github.head_ref || github.ref_name }}
+  BUILD_TAG: ${{ github.event.pull_request.head.sha || github.sha }}
+  JIRA_PROJECT_KEY: "MM"
+  MM_TEST_SERVER_URL: ${{ inputs.MM_TEST_SERVER_URL || secrets.MM_DESKTOP_E2E_SERVER_URL }}
+  MM_TEST_USER_NAME: ${{ inputs.MM_TEST_USER_NAME || secrets.MM_DESKTOP_E2E_USER_NAME }}
+  MM_TEST_PASSWORD: ${{ inputs.MM_TEST_PASSWORD || secrets.MM_DESKTOP_E2E_USER_CREDENTIALS }}
+  PULL_REQUEST: "https://github.com/mattermost/desktop/pull/${{ github.event.number }}"
+  ZEPHYR_ENVIRONMENT_NAME: "Desktop app"
+  ZEPHYR_FOLDER_ID: "12413253"
+  TEST_CYCLE_LINK_PREFIX: ${{ secrets.MM_DESKTOP_E2E_TEST_CYCLE_LINK_PREFIX }}
+  AWS_ACCESS_KEY_ID: ${{ secrets.MM_DESKTOP_E2E_AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.MM_DESKTOP_E2E_AWS_SECRET_ACCESS_KEY }}
+  AWS_REGION: "us-east-1"
+  WEBHOOK_URL: ${{ secrets.MM_DESKTOP_E2E_WEBHOOK_URL }}
+  ZEPHYR_API_KEY: ${{ secrets.MM_DESKTOP_E2E_ZEPHYR_API_KEY }}
+  REPORT_LINK: "none"
+
+jobs:
+  e2e:
+    runs-on: ${{ inputs.runs-on }}
+    defaults:
+      run:
+        shell: bash
+    outputs:
+      COMMENT_BODY: ${{ steps.analyze-flaky-tests.outputs.COMMENT_BODY }}
+    steps:
+      - name: e2e/set-required-variables
+        id: variables
+        run: |
+          if [ "${{ github.event_name }}" == "pull_request" ]; then
+            echo "BUILD_SUFFIX=desktop-pr-linux" >> $GITHUB_ENV
+            echo "TYPE=PR" >> $GITHUB_ENV
+
+          elif [ "${{ github.event_name }}" == "release" ]; then
+            echo "BUILD_SUFFIX=desktop-release-linux" >> $GITHUB_ENV
+            echo "TYPE=RELEASE" >> $GITHUB_ENV
+            echo "ZEPHYR_ENABLE=true" >> $GITHUB_ENV
+            echo "ZEPHYR_FOLDER_LINUX_REPORT=12358649" >> $GITHUB_ENV
+
+          elif [ "${{ github.event_name }}" == "workflow_dispatch" ] && ! ${{ inputs.nightly }}; then
+            echo "BUILD_SUFFIX=desktop-manual-trigger-linux" >> $GITHUB_ENV
+            echo "TYPE=MANUAL" >> $GITHUB_ENV
+
+          elif [ "${{ github.event_name }}" == "push" ] && [ "${{ github.ref }}" == "refs/heads/master" ]; then
+            echo "BUILD_SUFFIX=desktop-master-push-linux" >> $GITHUB_ENV
+            echo "TYPE=MASTER" >> $GITHUB_ENV
+            echo "ZEPHYR_ENABLE=true" >> $GITHUB_ENV
+
+          elif ${{ inputs.nightly }}; then
+            echo "BUILD_SUFFIX=desktop-nightly-linux" >> $GITHUB_ENV
+            echo "TYPE=NIGHTLY" >> $GITHUB_ENV
+            echo "ZEPHYR_ENABLE=true" >> $GITHUB_ENV
+            echo "ZEPHYR_FOLDER_LINUX_REPORT=12363689" >> $GITHUB_ENV
```

Comment:
@toninis just 1 change
this Line 77 and 92
have different folder values based on OS. Maybe this can also be set up to post to diff Folders based on OS

when TYPE= Nightly 
ZEPHYR_FOLDER_LINUX_REPORT=12363689
ZEPHYR_FOLDER_MACOS_REPORT=12363687
ZEPHYR_FOLDER_WIN_REPORT=12363690

when TYPE= Nightly 
ZEPHYR_FOLDER_LINUX_REPORT=12358649
ZEPHYR_FOLDER_MACOS_REPORT=12358650
ZEPHYR_FOLDER_WIN_REPORT=12358651

Reply:
@yasserfaraazkhan I added the variables . 
A couple of questions: 
- When does the `github.event_name == release` triggered ? 
- Do we need any extra configuration Zephyr configuration for the Compatibility Matrix Testing ?

Reply:
@toninis 
1. I kept is as a placeholder to add conditions for running the tests when release branch is cut. I didn't make any progress yet 😅 

2. Nothing more needed for Zephyr.
the reports will be collected in below folders.
<img width="905" alt="Screenshot 2024-03-22 at 3 02 33 PM" src="https://github.com/mattermost/desktop/assets/15979783/32364334-264a-407e-8248-06eadd779f71">